forge/src/main/java/org/openjdk/skara/forge/HostedRepository.java

```diff
@@ -87,6 +87,17 @@ PullRequest createPullRequest(HostedRepository target,
      * Returns contents of the file, if the file does not exist, returns Optional.empty().
      */
     Optional<String> fileContents(String filename, String ref);
+
+    /**
+     * Writes new contents to a file in the repo by creating a new commit.
+     * @param content New file content to write, always replacing existing content
+     * @param filename Name of file inside repository to write to
+     * @param branch Branch to add commit on top of
+     * @param message Commit message
+     * @param authorName Name of author and committer for commit
+     * @param authorEmail Email of author and committer for commit
+     */
+    void writeFileContents(String content, String filename, Branch branch, String message, String authorName, String authorEmail);
```

Comment:
Maybe not a big deal, but I thought the symmetry with `fileContents` was a bit lost when `String filename` was not the first argument to this method.

Reply:
Sure, makes sense.